ui/README.md

```diff
@@ -34,19 +24,51 @@ PORT=3001 npm start  # 3001 is the port number. Save port 3000
                      # for the development API server.
 ```
 
-A web browser should open automatically to open the app. You 
-won't see much interesting at first, as no PDF has been 
-loaded. Load a PDF by entering the following URL in the URL 
-bar: 
-`http://localhost:3001/?file=https://arxiv.org/pdf/0801.4750.pdf`.
+A web browser should open automatically to open the app. You
+won't see much interesting at first, as no PDF has been
+loaded. Load a PDF by visiting [this URL](http://localhost:3001/?file=https://arxiv.org/pdf/0801.4750.pdf).
 
-You can in theory provide the URL for any PDF as the value 
-of the `file` query parameter. That said, at the time this 
-README was written, there was only a few PDFs for which the 
+You can in theory provide the URL for any PDF as the value
+of the `file` query parameter. That said, at the time this
+README was written, there was only a few PDFs for which the
 augmented reading experience is available.
 
-If you are testing out the application, you should also make 
-sure that the API server is running on port 3000. See the 
-README in the `api/` sibling directory for instructions.
+## Architecture
+
+We leverage [Mozilla's PDF.js](https://mozilla.github.io/pdf.js/)
+for rendering the PDF in a browser. We maintain a [fork](https://github.com/allenai/scholar-reader-pdfjs)
+with several changes specific to our application.  The fork is
+linked to the code as a submodule.
+
+The code in `public/` is produced from our fork, and serves
+as a base into which our additional interface elements are
+rendered. You can regenerate this code and overwrite the copies
+in `public/` by executing these commands:
+
+```
+git submodule init
+git submodule update
+bash bin/build_pdf.js.sh
+```
+
+To apply the effect, commit the resulting changes.
```

Comment:
What do you mean by "apply the effect"?

Reply:
Running the script updates the files in `public/`. This is just noting that the user needs to commit them.

That said I think that's implicit. Once the operator executes `git status` the impact of `bin/build_pdf.js.sh` will be clear.

I'll just omit this from the docs.